File: xy11310/app/models/delivery.py
from sqlalchemy import Column, Integer, String, Text, DateTime, Date, ForeignKey, Enum
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base
import enum


class DeliveryStatus(str, enum.Enum):
    PENDING = "pending"
    DISPATCHED = "dispatched"
    DELIVERED = "delivered"
    FAILED = "failed"
    RETURNED = "returned"


class Delivery(Base):
    __tablename__ = "deliveries"

    id = Column(Integer, primary_key=True, index=True)
    delivery_batch_id = Column(String(100), index=True)
    
    elderly_id = Column(Integer, ForeignKey("elderly.id"), nullable=False)
    meal_allocation_id = Column(Integer, ForeignKey("meal_allocations.id"))
    
    delivery_date = Column(Date, nullable=False, index=True)
    meal_type = Column(String(20))
    
    route = Column(String(100))
    sequence = Column(Integer)
    
    status = Column(Enum(DeliveryStatus), default=DeliveryStatus.PENDING)
    
    delivered_by = Column(String(100))
    delivered_at = Column(DateTime(timezone=True))
    
    received_by = Column(String(100))
    signature = Column(String(200))
    
    failure_reason = Column(Text)
    
    temperature = Column(String(50))
    packaging_condition = Column(String(100))
    
    notes = Column(Text)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    elderly = relationship("Elderly")
    meal_allocation = relationship("MealAllocation")

File: xy11308/src/entities/Delivery.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn } from 'typeorm';
import { MealAssignment } from './MealAssignment';

export enum DeliveryStatus {
  PENDING = 'pending',
  IN_TRANSIT = 'in_transit',
  DELIVERED = 'delivered',
  FAILED = 'failed',
  RETURNED = 'returned'
}

@Entity('deliveries')
export class Delivery {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  assignmentId: string;

  @Column({ type: 'simple-enum', enum: DeliveryStatus, default: DeliveryStatus.PENDING })
  status: DeliveryStatus;

  @Column({ nullable: true })
  deliveryPerson: string;

  @Column({ type: 'simple-array', nullable: true })
  deliveryRoute: string[];

  @Column({ nullable: true })
  estimatedDeliveryTime: Date;

  @Column({ nullable: true })
  actualDeliveryTime: Date;

  @Column({ nullable: true })
  recipientName: string;

  @Column({ nullable: true })
  recipientSignature: string;

  @Column({ nullable: true })
  failureReason: string;

  @Column({ nullable: true })
  notes: string;

  @Column()
  createdBy: string;

  @Column()
  createdByRole: string;

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;

  @ManyToOne(() => MealAssignment)
  @JoinColumn({ name: 'assignmentId' })
  assignment: MealAssignment;
}